streams/src/main/java/org/apache/kafka/streams/processor/TaskMetadata.java

```diff
@@ -27,10 +27,12 @@
 
 /**
  * Represents the state of a single task running within a {@link KafkaStreams} application.
+ * @deprecated since 3.0, not intended for public use, use {@link org.apache.kafka.streams.TaskMetadata} instead.
  */
+@Deprecated
 public class TaskMetadata {
 
-    private final TaskId taskId;
+    private final String taskId;
```

Comment:
This reverts TaskMetadata to the state before KIP-740. As this TaskMetadata is deprecated, no need to do any type change here.